src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java

```diff
@@ -0,0 +1,192 @@
+
+		User user = userDao.findByAccountName(accountName);
+
+		// prepare set of authorities
+		Set<GrantedAuthority> grantedAuthorities = new HashSet<GrantedAuthority>();
+
+		String encryptedPassword = null;
+
+		if (user == null) {
+			LOG.warn("No user for account name '" + accountName + "' could be found.");
+			throw new UsernameNotFoundException(exceptionMessage);
+		} else if (!user.isActive()) {
+			LOG.warn("The user with the account name '" + accountName + "' is not active.");
+			throw new DisabledException(exceptionMessage);
+		} else {
+
+			encryptedPassword = user.getPassword();
+
+			// check if rawPassword matches the hash from db
+			if (passwordEncoder.matches(rawPassword, encryptedPassword)) {
+
+				Set<Role> allUserRoles = getAllUserRoles(user);
+
+				// create granted authorities for the security context
+				for (Role role : allUserRoles) {
+					grantedAuthorities.add(new SimpleGrantedAuthority(role.getName()));
+				}
+
+			} else {
+				LOG.warn("The given password for user '" + accountName + "' does not match.");
+				throw new BadCredentialsException(exceptionMessage);
+			}
+
+		}
+
+		// Create corresponding token to forward in Spring Security's filter
+		// chain. We will use the SHOGun2 user as the principal.
+		Authentication authResult = null;
+		if (grantedAuthorities.isEmpty()) {
+			// if the user has no authorities, we will build the
+			// UsernamePasswordAuthenticationToken without authorities, which
+			// leads to an unauthenticated user, i.e. isAuthenticated() of
+			// authenticationToken will return false afterwards.
+			LOG.warn("The user '" + accountName + "' has no authorities and will thereby NOT be authenticated.");
+			authResult = new UsernamePasswordAuthenticationToken(user, encryptedPassword);
+		} else {
+			// if we pass some grantedAuthorities, isAuthenticated() of
+			// authenticationToken will return true afterwards
+			authResult = new UsernamePasswordAuthenticationToken(user, encryptedPassword, grantedAuthorities);
+
+			LOG.debug("The user '" + accountName
+					+ "' got the following (explicit) roles: "
+					+ StringUtils.join(getRawRoleNames(grantedAuthorities), ", "));
+		}
+
+		final boolean isAuthenticated = authResult.isAuthenticated();
+		final String authLog = isAuthenticated ? "has succesfully" : "has NOT";
+		LOG.info("The user '" + accountName + "' " + authLog  + " been authenticated.");
+
+		return authResult;
+	}
+
+	/**
+	 *
+	 * @see org.springframework.security.authentication.AuthenticationProvider#supports
+	 *      (java.lang.Class)
+	 */
+	@Override
+	public boolean supports(Class<?> authentication) {
+		return (UsernamePasswordAuthenticationToken.class
+				.isAssignableFrom(authentication));
+	}
+
+	/**
+	 * @param user
+	 * @param userGroups
+	 * @return
+	 */
+	private Set<Role> getAllUserRoles(User user) {
+		Set<Role> allUserRoles = new HashSet<Role>();
+
+		if (user != null) {
+			List<UserGroupRole> result = userGroupRoleDao.findUserRoles(user);
```

Comment:
I think you could also use the `findAllWhereFieldEquals` from SHOGun, i.e. `findAllWhereFieldEquals('user', user)`:

https://github.com/terrestris/shogun2/blob/master/src/shogun2-core/src/main/java/de/terrestris/shogun2/dao/GenericHibernateDao.java#L110